src/Sema.zig

```diff
@@ -39123,3 +38907,77 @@ const loadComptimePtr = @import("Sema/comptime_ptr_access.zig").loadComptimePtr;
 const ComptimeLoadResult = @import("Sema/comptime_ptr_access.zig").ComptimeLoadResult;
 const storeComptimePtr = @import("Sema/comptime_ptr_access.zig").storeComptimePtr;
 const ComptimeStoreResult = @import("Sema/comptime_ptr_access.zig").ComptimeStoreResult;
+
+/// Convenience function that looks 2 levels deep into `std.builtin`.
+fn getBuiltinInnerAsInst(
+    sema: *Sema,
+    block: *Block,
+    src: LazySrcLoc,
+    outer_name: []const u8,
+    inner_name: []const u8,
+) !Air.Inst.Ref {
+    const outer_ty = try sema.getBuiltinType(outer_name);
+    const inner_val = try getInnerValue(sema, block, src, outer_ty, inner_name);
+    return Air.internedToRef(inner_val.toIntern());
+}
+
+/// Convenience function that looks 2 levels deep into `std.builtin`.
+fn getBuiltinInnerType(
+    sema: *Sema,
+    block: *Block,
+    src: LazySrcLoc,
+    outer_name: []const u8,
+    inner_name: []const u8,
+) !Type {
+    const outer_ty = try sema.getBuiltinType(outer_name);
+    return getInnerType(sema, block, src, outer_ty, inner_name);
+}
+
+fn getInnerType(
+    sema: *Sema,
+    block: *Block,
+    src: LazySrcLoc,
+    outer_ty: Type,
+    inner_name: []const u8,
+) !Type {
+    const inner_val = try getInnerValue(sema, block, src, outer_ty, inner_name);
+    return inner_val.toType();
+}
+
+fn getInnerValue(
+    sema: *Sema,
+    block: *Block,
+    src: LazySrcLoc,
+    outer_ty: Type,
+    inner_name: []const u8,
+) !Value {
+    const pt = sema.pt;
+    const zcu = pt.zcu;
+    const ip = &zcu.intern_pool;
+    const gpa = sema.gpa;
+    const nav = try sema.namespaceLookup(
+        block,
+        src,
+        outer_ty.getNamespaceIndex(zcu),
+        try ip.getOrPutString(gpa, pt.tid, inner_name, .no_embedded_nulls),
+    ) orelse return sema.fail(block, src, "std.builtin missing {s}", .{inner_name});
+    try sema.ensureNavResolved(src, nav);
+    return Value.fromInterned(ip.getNav(nav).status.resolved.val);
```

Comment:
@mlugg should this be calling `analyzeNavVal` instead?

Reply:
Yes, but also, see `Sema.namespaceLookupVal` which just wraps these two calls.

Don't worry too much about incremental dependencies for `std.builtin` stuff at the minute -- incremental updates to `std.builtin` are very broken as-is, this is on my todo list for incremental.